### PR TITLE
Inject mozPipelineMetadata to generated Glean json schemas

### PR DESCRIPTION
**Pull request by jklukas**
Makes Glean JSON schemas parallel to hand-written JSON schemas. See
https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/579

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ RUN pip install -r requirements/test_requirements.txt
 ADD . ${HOME}/mozilla-schema-generator
 ENV PATH $PATH:${HOME}/mozilla-schema-generator/bin
 
-RUN pip install -e ${HOME}/mozilla-schema-generator
+RUN pip install --no-dependencies -e ${HOME}/mozilla-schema-generator
```
Comment (jklukas) [inline Dockerfile:41]: This is an unrelated efficiency boost. We already install dependencies based on requirements.txt before this line, so we can save some time by avoiding validating dependencies here.

File: bin/alias_schemas [hunk 2]
```diff
@@ -57,7 +57,7 @@ def main(aliases_path, base_dir):
                 fname = source.name.replace(source_doctype, dest_doctype)
                 dest = dest_dir / fname
 
-                print(f"Aliasing {dest} to {source}")
+                print(f"Aliasing {source} to {dest}")
```
Comment (jklukas) [inline bin/alias_schemas:60]: Unrelated bug fix.

File: mozilla_schema_generator/__main__.py [hunk 3]
```diff
@@ -214,16 +214,21 @@ def generate_glean_pings(config, out_dir, split, pretty, repo, generic_schema):
...
+    json_dump_args = {
+        'cls': SchemaEncoder
+    }
     if pretty:
-        json_dump_args['indent'] = 4
-        json_dump_args['separators'] = (',', ':')
+        json_dump_args.update({
+            'indent': 4,
+            'separators': (',', ':'),
+            'sort_keys': True,
```
Comment (jklukas) [inline mozilla_schema_generator/__main__.py:230]: Adding `sort_keys` here makes sure that `mozPipelineMetadata` appears near the top of the generated schemas, in line with how hand-written schemas from mps look.

File: mozilla_schema_generator/__main__.py [hunk 4]
```diff
@@ -214,16 +214,21 @@ def generate_glean_pings(config, out_dir, split, pretty, repo, generic_schema):
 
 
 def write_schema(repo, repo_id, config, out_dir, split, pretty, generic_schema):
-    schema_generator = GleanPing(repo)
+    schema_generator = GleanPing(repo, repo_id)
```
Comment (jklukas) [inline mozilla_schema_generator/__main__.py:217]: We need GleanPing to keep track of the repo_id (app_id) now since the generated metadata depends on it for the bq_dataset_family param.

**Comment by jklukas** [comment]
Looks like I need to update tests

**Comment by jklukas** [comment]
Tests are now passing, and you can see the effects of these logic changes in the following commit to `test-generated-schemas`:

https://github.com/mozilla-services/mozilla-pipeline-schemas/commit/7a366cf2ef9dbe9eda6842ec40bf84c8badb0013

File: bin/alias_schemas (hunk 2 above)
Comment (fbertsch) [reply bin/alias_schemas:60]: Hah, I remember writing this and being undecided about the order.

File: tests/test_glean.py [hunk 5]
```diff
@@ -67,11 +67,17 @@ def test_generic_schema(self, glean, config):
 
         final_schemas = {k: schemas[k][0].schema for k in schemas}
         for name, schema in final_schemas.items():
+            meta = schema.pop("mozPipelineMetadata")
```
Comment (fbertsch) [inline tests/test_glean.py:70]: `print_and_test` makes missing/altered elements in JSON much easier to understand on failure. Can We instead explicitly update the `generic_schema` to include this section?

File: bin/alias_schemas (hunk 2 above)
Comment (jklukas) [reply bin/alias_schemas:60]: I was definitely interpreting this as basically meaning "I've copied X source file to Y destination". But I can now see how you might think about aliasing the opposite way.

File: tests/test_glean.py (hunk 5 above)
Comment (jklukas) [reply tests/test_glean.py:70]: I'll look into whether we can refactor this.

Comment (fbertsch) [reply tests/test_glean.py:70]: Just to be clear - I don't see a great way to update the class function. Instead just add this blob to `generic_schema` in this test, and compare it all using `print_and_test`.

File: bin/alias_schemas (hunk 2 above)
Comment (fbertsch) [reply bin/alias_schemas:60]: Yeah, probably just needs a bit more explanatory text.
